### PR TITLE
[NSTK-2164] Release 6.0.4

### PR DESCRIPTION
**Pull request by haibinxie**
- Ephemeral volume support.
- CSI 1.3 compliance.
- Support HostName using FQDN.
- Improve volume provisioning performance when topology is enabled
- Add default cockroach DB pod memory limit to 1GB
- Fix volume attach failure when active cluster is enabled.
- Fix failure when image contains port as suffix of repository.

### CODE REVIEW

File: mirror_pso_containers.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ quay.io/k8scsi/livenessprobe:v2.0.0
 quay.io/k8scsi/csi-node-driver-registrar:v1.3.0
 purestorage/cockroach-operator:v1.0.3
```
Comment (pure-jliao) [inline mirror_pso_containers.sh:18]: update cockroach-operator and dbdeployer to v1.0.4 and k8s to v6.0.4

File: pure-pso/README.md [hunk 2]
```diff
@@ -271,8 +271,8 @@ Strict attention must be paid to the versions of image you provide locally as PS
 | quay.io/k8scsi/csi-node-driver-registrar | v1.3.0  |
 | purestorage/cockroach-operator           | v1.0.3  |
```
Comment (pure-jliao) [inline pure-pso/README.md:272]: cockroach-operator and dbdeployer to v1.0.4

File: pure-pso/values.yaml [hunk 3]
```diff
@@ -99,7 +99,7 @@ images:
       pullPolicy: Always
     psctl:
       name: purestorage/psctl
-      tag: v1.0.1
+      tag: v1.0.2
```
Comment (pure-jliao) [inline pure-pso/values.yaml:102]: k8s, cockroach-operator and dbdeployer need to be updated in values.yaml

**Review by pure-jliao — CHANGES_REQUESTED** [review]
we need to put the notes in CHANGELOG.md, then update all the tag versions in mirror_pso_containers.sh, pure-pso/README.md, and values.yaml, finally rerun update.sh to update the PR.

**Review by pure-jliao — APPROVED** [review]
LGTM! don't forget to squash commits!

**Review by dsupure — APPROVED** [review]
LGTM!

**Comment by david-enli** [comment]
just a reminder for everyone, we need to take out FQDN for now

**Review by david-enli — CHANGES_REQUESTED** [review]
just a reminder for us to remove the FQDN line

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+
+* CSI 1.3 compliance.
+* Improve volume provisioning performance when topology is enabled.
+* Add default cockroach DB pod memory limit to 1GB.
+
+#### Bug Fixes:
+
+* Support HostName using FQDN.
+* Fix volume attach failure when active cluster is enabled.
+* Fix failure when image contains port as suffix of repository.
```
Comment (pure-jliao) [inline CHANGELOG.md:19]: Fix filesystem / raw block within pod doesn't change after volume expansion.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,23 @@
...
+
+#### Major Features:
+
+* Ephemeral volume support.
+
+#### Enhancements:
+
+* CSI 1.3 compliance.
+* Improve volume provisioning performance when topology is enabled.
+* Add default cockroach DB pod memory limit to 1GB.
```
Comment (pure-jliao) [inline CHANGELOG.md:13]: Volume expansion no longer needs pod restart.

**Review by pure-jliao — COMMENTED** [review]
Could you help to add those in the changelog? thanks.

**Pull request merged**